sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
again the same test case won't pass. It gives the `ValueError` as:
```
  File "c:\users\dell\desktop\sympy\sympy\solvers\tests\test_solveset.py", line 882, in test_solveset
    assert solveset(0, domain=S.Reals) == S.Reals
  File "C:\Users\DELL\Desktop\sympy\sympy\solvers\solveset.py", line 865, in solveset
    multivariate equation.'''))
ValueError:
The independent variable must be specified for a multivariate
equation.

 tests finished: 345 passed, 21 skipped, 32 expected to fail, 1 exceptions,
in 328.50 seconds
DO *NOT* COMMIT!
```
I guess the condition has to be checked only after coming into the `if block` of Line 848, otherwise the case `solveset(0, domain=S.Reals)` won't be checked.